configure.ac

```diff
@@ -246,6 +246,18 @@ AC_ARG_ENABLE([memory-checking],
 AC_MSG_CHECKING([whether to enable memory checking])
 AC_MSG_RESULT([$enable_memory_checking])
 
+AC_ARG_ENABLE([valgrind],
+    [AS_HELP_STRING([--enable-valgrind], [add valgrind extensions to GASMAN])],
+    [AC_DEFINE([MEMORY_CANARY], [1], [define if build with valgrind extensions])],
+    [enable_valgrind=no]
+    )
+AC_MSG_CHECKING([whether to add valgrind extensions to GASMAN])
```

Comment:
Optional: Instead of "add" here and above, I'd use "enable", to match the name of the option. Also, to me "add" has the wrong meaning here, but that's probably just me being a non-native speaker.